src/Cassandra/DataStax/Cloud/CustomCACertificateValidator.cs

```diff
@@ -145,6 +148,55 @@ public bool Validate(X509Certificate cert, X509Chain chain, SslPolicyErrors erro
             return valid;
         }
 
+        private bool ValidateName(string name)
+        {
+            if (name == null || _hostname == null || !name.StartsWith("*."))
+            {
+                if (name?.ToLowerInvariant() == _hostname?.ToLowerInvariant())
+                {
+                    return true;
+                }
+            }
+            else if (name.StartsWith("*."))
+            {
+                name = name.Remove(0, 1);
+                if (_hostname.EndsWith(name, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private IEnumerable<string> GetSubjectAlternativeNames(X509Certificate2 cert)
+        {
+            var result = new List<string>();
+
+            var subjectAlternativeName = cert.Extensions.Cast<X509Extension>()
+                .Where(n => n.Oid.Value == "2.5.29.17") //n.Oid.FriendlyName=="Subject Alternative Name")
+                .Select(n => new AsnEncodedData(n.Oid, n.RawData))
+                .Select(n => n.Format(true))
+                .FirstOrDefault();
+
+            if (subjectAlternativeName != null)
+            {
+                var alternativeNames = subjectAlternativeName.Split(new[] { "\r\n", "\r", "\n" }, StringSplitOptions.None);
+
+                foreach (var alternativeName in alternativeNames)
+                {
+                    var groups = Regex.Match(alternativeName, @"^(.*)=(.*)").Groups; // @"^DNS Name=(.*)").Groups;
```

Comment:
Is this intended to... replace the DnsNameRegex referenced above?  Seems like it's trying to do the same thing with a different regex.

Reply:
Ahh I had to try multiple solutions and forgot about those constants... fixed